test/scripts/cases/_common.sh

```
@@ -0,0 +1,194 @@
+
+# Function to check if a command exists
+command_exists() {
+    command -v "$1" &> /dev/null
+}
+
+# Function to create results directory with timestamp
+create_results_dir() {
+    local prefix="${1:-test-results}"
+    local results_base="$PROJECT_ROOT/test/results"
+    mkdir -p "$results_base"
+    local results_dir="$results_base/$prefix-$(date +%Y%m%d-%H%M%S)"
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)